src/WebJobs.Script/OutOfProc/Http/HttpFunctionInvokeDispatcher.cs

```diff
@@ -0,0 +1,200 @@
+
+        public HttpFunctionInvokeDispatcher(IOptions<ScriptJobHostOptions> scriptHostOptions,
+            IMetricsLogger metricsLogger,
+            IEnvironment environment,
+            IScriptJobHostEnvironment scriptJobHostEnvironment,
+            IScriptEventManager eventManager,
+            ILoggerFactory loggerFactory,
+            IHttpInvokerChannelFactory httpInvokerChannelFactory,
+            IHttpInvokerService httpInvokerService,
+            IOptions<LanguageWorkerOptions> languageWorkerOptions,
+            IJobHostLanguageWorkerChannelManager jobHostLanguageWorkerChannelManager)
+        {
+            _metricsLogger = metricsLogger;
+            _scriptOptions = scriptHostOptions.Value;
+            _environment = environment;
+            _scriptJobHostEnvironment = scriptJobHostEnvironment;
+            _eventManager = eventManager;
+            _workerConfigs = languageWorkerOptions.Value.WorkerConfigs;
+            _logger = loggerFactory.CreateLogger<HttpFunctionInvokeDispatcher>();
+            _httpInvokerChannelFactory = httpInvokerChannelFactory;
+            _httpInvokerService = httpInvokerService;
+
+            State = FunctionDispatcherState.Default;
+
+            _workerErrorSubscription = _eventManager.OfType<HttpWorkerErrorEvent>()
+               .Subscribe(WorkerError);
+            _workerRestartSubscription = _eventManager.OfType<HttpWorkerRestartEvent>()
+               .Subscribe(WorkerRestart);
+        }
+
+        public FunctionDispatcherState State { get; private set; }
+
+        internal ConcurrentStack<HttpWorkerErrorEvent> LanguageWorkerErrors => _languageWorkerErrors;
+
+        internal async void InitializeJobhostLanguageWorkerChannelAsync()
+        {
+            await InitializeJobhostLanguageWorkerChannelAsync(0);
+        }
+
+        internal Task InitializeJobhostLanguageWorkerChannelAsync(int attemptCount)
+        {
+            // TODO: Add process managment for http invoker
+            _httpInvokerChannel = _httpInvokerChannelFactory.CreateHttpInvokerChannel(_scriptOptions.RootScriptPath, _metricsLogger, attemptCount);
+            _httpInvokerChannel.StartWorkerProcessAsync().ContinueWith(workerInitTask =>
+                 {
+                     if (workerInitTask.IsCompleted)
+                     {
+                         _logger.LogDebug("Adding http invoker channel. workerId:{id}", _httpInvokerChannel.Id);
+                         // TODO: wait for status api ok
+                         State = FunctionDispatcherState.Initialized;
+                     }
+                     else
+                     {
+                         _logger.LogWarning("Failed to start http invoker process. workerId:{id}", _httpInvokerChannel.Id);
+                     }
+                 });
+            return Task.CompletedTask;
+        }
+
+        public bool IsSupported(FunctionMetadata functionMetadata, string workerRuntime)
+        {
+            return true;
```

Comment:
Is there a "to do" here?

Reply:
good catch! thanks. Runtime check is not needed in the dispatcher. Cleaned up interface